packages/lambda-powertools-middleware-log-timeout/__tests__/index.js

```diff
@@ -0,0 +1,68 @@
+const consoleLog = jest.spyOn(global.console, 'log')
+
+const middy = require('middy')
+const logTimeoutMiddleware = require('../index')
+
+const delay = (millis) => new Promise((resolve) => setTimeout(resolve, millis))
+
+beforeEach(() => {
+  consoleLog.mockClear()
+})
+
+const invokeSuccessHandler = async () => {
+  const handler = middy(async () => {
+  })
+  handler.use(logTimeoutMiddleware())
+
+  await handler({}, { awsRequestId: 'test-id' }, () => {})
+}
+
+const invokeTimedOutHandler = async (event, awsRequestId) => {
+  const context = {
+    awsRequestId,
+    getRemainingTimeInMillis: () => 20
+  }
+
+  const handler = middy(async () => {
+    await delay(20)
+  })
+  handler.use(logTimeoutMiddleware())
+
+  await new Promise((resolve) => {
+    handler(event, context, resolve)
+  })
+}
+
+const errorLogWasWritten = (f) => {
+  expect(consoleLog).toBeCalled()
+  const log = JSON.parse(consoleLog.mock.calls[0])
+  expect(log.sLevel).toBe('ERROR')
+  expect(log.level).toBe(50)
+  expect(log.message).toBe('invocation timed out')
+
+  f(log)
+}
+
+describe('Log timeout middleware', () => {
+  describe('when function finishes successfully', () => {
+    it('does not log anything', async () => {
+      await invokeSuccessHandler()
+      expect(consoleLog).not.toBeCalled()
+    })
+  })
+
+  describe('when function times out', () => {
```

Comment:
Can an additional test also be added to ensure that a custom threshold for `thresholdMillis` works when passed in as an arg?

Reply:
could do, but the timing etc. might be tricky - timer event is not guaranteed to fire exactly on time, and in any case we'd need to check the timing of the log message, any thoughts on how we'd do that, i.e. check that the log message was recorded X ms before the function timed out?

Reply:
https://jestjs.io/docs/en/timer-mocks#advance-timers-by-time will work - you can advance a mock timer by the exact amount of time when the log should be written and assert that

Reply:
added tests for custom thresholds